src/client.ts

```diff
@@ -465,36 +719,36 @@ export class MatrixClient extends EventEmitter {
     protected canSupportVoip = false;
     protected peekSync: SyncApi = null;
     protected isGuestAccount = false;
-    protected ongoingScrollbacks: {[roomId: string]: {promise?: Promise<any>, errorTs?: number}} = {}; // TODO: Types
+    protected ongoingScrollbacks: {[roomId: string]: {promise?: Promise<Room>, errorTs?: number}} = {};
     protected notifTimelineSet: EventTimelineSet = null;
     protected cryptoStore: CryptoStore;
-    protected verificationMethods: string[];
+    protected verificationMethods: VerificationMethod[];
     protected fallbackICEServerAllowed = false;
     protected roomList: RoomList;
     protected syncApi: SyncApi;
     public pushRules: any; // TODO: Types
     protected syncLeftRoomsPromise: Promise<Room[]>;
     protected syncedLeftRooms = false;
     protected clientOpts: IStoredClientOpts;
-    protected clientWellKnownIntervalID: number;
+    protected clientWellKnownIntervalID: NodeJS.Timeout;
     protected canResetTimelineCallback: ResetTimelineCallback;
 
     // The pushprocessor caches useful things, so keep one and re-use it
     protected pushProcessor = new PushProcessor(this);
 
     // Promise to a response of the server's /versions response
     // TODO: This should expire: https://github.com/matrix-org/matrix-js-sdk/issues/1020
-    protected serverVersionsPromise: Promise<any>;
+    protected serverVersionsPromise: Promise<IServerVersions>;
 
     protected cachedCapabilities: {
         capabilities: ICapabilities;
         expiration: number;
     };
-    protected clientWellKnown: any;
-    protected clientWellKnownPromise: Promise<any>;
-    protected turnServers: any[] = []; // TODO: Types
+    protected clientWellKnown: IClientWellKnown;
+    protected clientWellKnownPromise: Promise<IClientWellKnown>;
+    protected turnServers: ITurnServer[] = [];
     protected turnServersExpiry = 0;
-    protected checkTurnServersIntervalID: number;
+    protected checkTurnServersIntervalID: NodeJS.Timeout;
```

Comment:
This is what we do in all of react-sdk

`declare function clearTimeout(timeoutId: NodeJS.Timeout): void;`

otherwise we have to cast on both sides and it is messy. We should be consistent and fix is holistically somehow.

Reply:
well, as I've said elsewhere: just because we do it elsewhere doesn't mean it's a good example :P 

We should indeed figure out this timer issue and fix it, but this might not be the PR that does it. In any case, making the problem worse is not desirable at the moment.

Reply:
oh lol, looks like it is as simple as using `window.setTimeout` - wonder if we can coerce the global.setTimeout to also have that signature in global.d.ts